src/base/bittorrent/session.cpp

```diff
@@ -381,7 +381,14 @@ Session::Session(QObject *parent)
     , m_diskCacheSize(BITTORRENT_SESSION_KEY(u"DiskCacheSize"_qs), -1)
     , m_diskCacheTTL(BITTORRENT_SESSION_KEY(u"DiskCacheTTL"_qs), 60)
     , m_diskQueueSize(BITTORRENT_SESSION_KEY(u"DiskQueueSize"_qs), (1024 * 1024))
+#if (LIBTORRENT_VERSION_NUM >= 20006)
```

Comment:
There are really two kinds of changes here:
1. Splitting UseOSCache into DiskIOReadMode+DiskIOWriteMode. This should be made in libtorrent version independent way.
2. Adding WriteThrough option to DiskIOWriteMode.

I would make the 2. also in libtorrent version independent way at the qBittorrent level for better compatibility so that WriteThrough option will effectively select DisableOSCache with incompatible libtorrent version.

Reply:
How’ll you deal with users claiming “write through” doesn’t work and seem to be on incompatible version.

Reply:
GUI: Don't display "Write through" option at all.

WebUI: Display it as "Write through (libtorrent 2.0.6+)"

When reading settings file or parsing API request just turn it into "Disable OS cache" as I suggested above.

Reply:
Done & done. Please check.